File: two-pointers/167-two-sum-ii.py
class Solution(object):
    def twoSum(self, numbers, target):
        l=0
        r=len(numbers)-1
        while l<r:
            C_sum=numbers[l]+numbers[r]
            if C_sum==target:
                return[l+1,r+1]
            elif C_sum<target:
                l+=1
            else:
                r-=1
        return[]

File: two-pointers/167-two-sum-ii.c
#include <stdlib.h>

int* twoSum(int* numbers, int numbersSize, int target, int* returnSize) {
    int* result = (int*)malloc(2 * sizeof(int));
    *returnSize = 2;
    
    int l = 0;
    int r = numbersSize - 1;
    
    while (l < r) {
        int total = numbers[l] + numbers[r];
        if (total == target) {
            result[0] = l + 1;
            result[1] = r + 1;
            return result;
        } else if (total < target) {
            l++;
        } else {
            r--;
        }
    }
    
    *returnSize = 0;
    free(result);
    return NULL;
}
